src/binding/c/request_api.txt

```diff
@@ -135,6 +135,16 @@ MPI_Test_cancelled:
 MPI_Test:
     .desc: Test for the completion of a request
     .docnotes: waitstatus
+{ -- early_return --
+    if (*request == MPI_REQUEST_NULL) {
+        *flag = 1;
+        MPIR_Status_set_empty(status);
+        if (status != MPI_STATUS_IGNORE) {
+            status->MPI_ERROR = MPI_SUCCESS;
+        }
```

Comment:
I'm not sure this part is actually necessary. Since this is a single request, the error is returned from the function. The standard says `status->MPI_ERROR` is undefined in this case, I believe.

Reply:
I think having a clean "empty" status won't hurt, necessary or not.

Reply:
BTW, is an empty status the same as `memset(status, 0, ...)`?

Reply:
Yes, our empty status is functionally a zero struct. We can convert the macro to do just that.

Reply:
> Yes, our empty status is functionally a zero struct. We can convert the macro to do just that.

Sorry, this was wrong. The `MPI_SOURCE` and `MPI_TAG` fields are not zero. They are set to `MPI_ANY_SOURCE` and `MPI_ANY_TAG`. All other fields are zero.